runtime/src/iree-amd-aie/aie_runtime/iree_aie_router.cc

```diff
@@ -43,295 +49,194 @@ int columns(const mlir::iree_compiler::AMDAIE::AMDAIEDeviceModel &deviceModel);
-                    connectionMatrix, inPortPktCount, maxPktStream) ==
-           std::tie(rhs.col, rhs.row, rhs.id, rhs.inPortId, rhs.outPortId,
-                    rhs.inPortToId, rhs.outPortToId, rhs.connectionMatrix,
-                    rhs.inPortPktCount, rhs.maxPktStream);
-  }
-};
 
-struct SwitchBoxConnectionEdge {
-  SwitchBoxNode &src;
-  SwitchBoxNode &target;
-  int maxCapacity;
-  StrmSwPortType bundle;
+enum class Connectivity { INVALID = 0, AVAILABLE = 1 };
+
+using SwitchboxConnect = struct SwitchboxConnect {
+  SwitchboxConnect() = default;
+  SwitchboxConnect(TileLoc tileLoc) : srcCoords(tileLoc), dstCoords(tileLoc) {}
+  SwitchboxConnect(TileLoc srcCoords, TileLoc dstCoords)
+      : srcCoords(srcCoords), dstCoords(dstCoords) {}
+
+  TileLoc srcCoords, dstCoords;
+  std::vector<Port> srcPorts;
+  std::vector<Port> dstPorts;
+  // connectivity between ports
+  std::vector<std::vector<Connectivity>> connectivity;
+  // weights of Dijkstra's shortest path
+  std::vector<std::vector<double>> demand;
+  // history of Channel being over capacity
+  std::vector<std::vector<int>> overCapacity;
+  // how many circuit streams are actually using this Channel
+  std::vector<std::vector<int>> usedCapacity;
+  // how many packet streams are actually using this Channel
+  std::vector<std::vector<int>> packetFlowCount;
+
+  // resize the matrices to the size of srcPorts and dstPorts
+  void resize() {
+    connectivity.resize(
+        srcPorts.size(),
+        std::vector<Connectivity>(dstPorts.size(), Connectivity::INVALID));
+    demand.resize(srcPorts.size(), std::vector<double>(dstPorts.size(), 0.0));
+    overCapacity.resize(srcPorts.size(), std::vector<int>(dstPorts.size(), 0));
+    usedCapacity.resize(srcPorts.size(), std::vector<int>(dstPorts.size(), 0));
+    packetFlowCount.resize(srcPorts.size(),
+                           std::vector<int>(dstPorts.size(), 0));
+  }
 
-  SwitchBoxConnectionEdge(SwitchBoxNode &src, SwitchBoxNode &target);
-};
+  // update demand at the beginning of each dijkstraShortestPaths iteration
+  void updateDemand() {
+    for (size_t i = 0; i < srcPorts.size(); i++) {
+      for (size_t j = 0; j < dstPorts.size(); j++) {
+        double history = DEMAND_BASE + OVER_CAPACITY_COEFF * overCapacity[i][j];
+        double congestion =
+            DEMAND_BASE + USED_CAPACITY_COEFF * usedCapacity[i][j];
+        demand[i][j] = history * congestion;
+      }
+    }
+  }
 
-// A node holds a pointer
-struct PathEndPointNode : PathEndPoint {
-  PathEndPointNode(SwitchBoxNode &sb, Port port)
-      : PathEndPoint{static_cast<SwitchBox>(sb), port}, sb(sb) {}
-  SwitchBoxNode &sb;
+  // inside each dijkstraShortestPaths interation, bump demand when exceeds
+  // capacity
+  void bumpDemand(size_t i, size_t j) {
+    if (usedCapacity[i][j] >= MAX_CIRCUIT_STREAM_CAPACITY) {
+      demand[i][j] *= DEMAND_COEFF;
+    }
+  }
 };
 
-struct FlowNode {
+using Flow = struct Flow {
```

Comment:
nit: I'm not a fan of typedefs for structs - I don't remember what it was but aie-rt makes things difficult for us becaues they do this everywhere.

Reply:
Yeah, I will change it.